engine_redis_test.go

```diff
@@ -605,34 +608,59 @@ func TestRedisEngineHandlePubSubMessage(t *testing.T) {
 	require.True(t, leaveHandlerCalled)
 }
 
+func BenchmarkRedisExtractPushData(b *testing.B) {
+	data := []byte(`__p1:16901:xght__\x12\nchat:index\x1aU\"\x0e{\"input\":\"__\"}*C\n\x0242\x12$37cb00a9-bcfa-4284-a1ae-607c7da3a8f4\x1a\x15{\"name\": \"Alexander\"}\"\x00`)
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		_, _, sp := extractPushData(data)
+		if sp.Offset != 16901 {
+			b.Fatal("wrong offset")
+		}
+		if sp.Epoch != "xght" {
+			b.Fatal("wrong epoch")
+		}
+	}
+	b.ReportAllocs()
```

Comment:
should this be on the top of the bench func?

Reply:
thanks, fixed - for some reason I thought it should always be called in the end of bench